File: 프로그래머스/lv2/42747. H－Index/H－Index.py
def solution(citations):
    count = [0] * (max(citations) + 1);
    for x in citations:
        count[x] += 1;
    
    answer = 0;
    for i in range(len(count)):
        less = sum(count[0: i+1]);
        more = sum(count[i:]);
        if i > more: break;
        answer = i;
    return answer

File: 프로그래머스/lv2/42747. H－Index/H－Index.js
function solution(citations) {
    const count = new Array(Math.max(...citations) + 1).fill(0);
    let answer = 0;
    citations.forEach(el => count[el] += 1)
    
    for(let i = 0; i < count.length; i++){
        let less = count.slice(0, i + 1).reduce((acc, cur) => acc + cur, 0);
        let more = count.slice(i).reduce((acc, cur) => acc + cur, 0);
        // if(less > more) return i - 1;
        if (i > more) break;
        
        answer = i;
        
        // if(less === more) return i;
    }
    
    return answer;
}